src/thumbnails.js

```diff
@@ -0,0 +1,70 @@
+// src/thumbnails.js
+//
+// Copyright (c) 2018 Endless Mobile Inc.
+//
+// Helper methods to make thumbnailing work in a flatpak
+// sandbox
+//
+
+const ChatboxPrivate = imports.gi.ChatboxPrivate;
+const Gio = imports.gi.Gio;
+const GLib = imports.gi.GLib;
+
+const PromiseHelpers = imports.gi.promiseHelpers
+
+function listDirectoryAsync(directory, cancellable) {
+    return promisifyGIO(ChatboxPrivate,
+                        'list_all_paths_in_directory_async',
+                        'list_all_paths_in_directory_finish',
+                        directory,
+                        'standard::name',
+                        Gio.FileQueryInfoFlags.NONE,
+                        cancellable);
+}
+
+function rewriteKeyFileInto(destinationDirectory,
```

Comment:
Arguments here should go `thumbailerPath`, `destinationDirectory`, `rewriteKeyPaths`